### PR TITLE
riscv: telink: update start.s

### PR DESCRIPTION
**Pull request by charlie-ai**
- disable vector mode for plic.

### CODE REVIEW

**Comment by haiwentelink** [comment]
when we use the driver or zigbee fw jump directly to matter fw , and find it will block on switching to main func , after that we find some differences in start.s in zephyr and other sdk , try disable vector mode in other sdk before XIP to matter , finally find it works .
         so I add the disable vector mode code into start.S, and test it , it also can works . the demo code in driver sdk src is :

       ```
 clear_csr(NDS_MMISC_CTL, BIT(1));
	__asm("lui	t0, 0xe4000");
	__asm("li   t1, 0x00");
	__asm("sw	t1, 0x0(t0)");，
```

**Comment by haiwentelink** [comment]
we use the driver or zigbee fw jump directly to matter fw , and find it will block on switching to main func , after that we find some differences in start.s in zephyr and other sdk , try disable vector mode in other sdk before XIP to matter ,  find it works .
     so I add the disable vector mode code into start.S, and test it , it also can works . the demo code in driver sdk src is :
 
```
         clear_csr(NDS_MMISC_CTL, BIT(1));
	__asm("lui	t0, 0xe4000");
	__asm("li   t1, 0x00");
	__asm("sw	t1, 0x0(t0)");
```

**Comment by s07641069** [comment]
@charlie-ai @haiwentelink 
Could we move new section under some config which will be disabled by default?
As I understand this is needed only for one Zigbee/Matter project.

@andriy-bilynskyy Could you provide you feedback and some short explanation of potential issues?

**Comment by andriy-bilynskyy** [comment]
Sure, vectored mode should be disabled under RTOSes
Relative to D25 core:
from AndesCoreTM D25 Data Sheet 1.3 2019-07-08 p.164
```
VEC_PLIC [1] Selects the operation mode of PLIC   RW 0
```
_Seems that core after reset is already in Regular mode_
Relative to PLIC:
from Gowin RiscV AE250 User Guide 01/21/2020 1.1E p. 168
21.5.2 Feature Enable Register
```
PREEMPT  [0] Preemptive priority interrupt enable RW 0
VECTORED [1] Vector mode enable                   RW 0
```
_Seems that PLIC after reset is already having features off_
But anyway seems beautiful - we are disabling to be fully sure...

File: soc/riscv/riscv-privilege/telink_b9x/start.S [hunk 1]
```diff
@@ -62,6 +62,17 @@ start:
...
+	/* Disable vector mode */
+	csrci  0x7d0, 2
+	/* Disable vector feture enable address */
+#if CONFIG_SOC_RISCV_TELINK_B91 || CONFIG_SOC_RISCV_TELINK_B92
+	lui t0 ,0xe4000
+#elif CONFIG_SOC_RISCV_TELINK_B95
+	lui t0 ,0xc4000
+#endif
+	li  t1 ,0x00
+	sw  t1 ,0x4(t0)
```
Comment (andriy-bilynskyy) [inline soc/riscv/riscv-privilege/telink_b9x/start.S]: Now PLIC base is in t0.
Feature enable register of PLIC has offset 0.
Why are we writing at offset 4 from PLIC base?
Seems like we set to zero source 1 priority instead feature register...

**Comment by andriy-bilynskyy** [comment]
https://github.com/telink-semi/zephyr/pull/266#issuecomment-2149486810
This comment provides correct code.

File: soc/riscv/riscv-privilege/telink_b9x/start.S (hunk 1 above)
Comment (haiwentelink) [reply soc/riscv/riscv-privilege/telink_b9x/start.S]: already changed , misunderstanding of the register , offset 0 means feature enable , offset 4 is the priority setting part .

**Review by andriy-bilynskyy — APPROVED** [review]

**Pull request merged**